src/service_objects/game_piece_action_resolvers/melee_attack_resolver.ts

```diff
@@ -90,15 +99,15 @@ export default async function resolveMeleeAttackAction(
     attackingGamePiece.gameId
   );
 
-  const snarkyGameState = new PhaseState(
-    Field(0),
-    Field(0),
-    startingGamePiecesTree.tree.getRoot(),
-    startingGamePiecesTree.tree.getRoot(),
-    startingGameArenaTree.tree.getRoot(),
-    startingGameArenaTree.tree.getRoot(),
-    playerPublicKey
-  );
+  const snarkyGameState = new PhaseState({
+    nonce: Field(0),
```

Comment:
I updated the contracts package to use this more explicit invocation.  Not everywhere yet, but I'm heading in that direction.